oxidation/libparsec_python/src/invite.rs

```diff
@@ -83,3 +85,460 @@ impl InvitationToken {
         }
     }
 }
+
+/*
+* SASCode
+*/
+
+#[pyclass]
+#[derive(PartialEq, Eq)]
+pub struct SASCode(pub parsec_api_types::SASCode);
+
+#[pymethods]
+impl SASCode {
+    #[new]
+    pub fn new(code: &str) -> PyResult<Self> {
+        match code.parse::<parsec_api_types::SASCode>() {
+            Ok(sas) => Ok(Self(sas)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    #[classmethod]
+    pub fn from_int(_cls: &PyType, num: u32) -> PyResult<Self> {
+        match parsec_api_types::SASCode::try_from(num) {
+            Ok(sas) => Ok(Self(sas)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    fn __richcmp__(&self, py: Python, other: &SASCode, op: CompareOp) -> PyObject {
+        match op {
+            CompareOp::Eq => (self.0.to_string() == other.0.to_string()).into_py(py),
```

Comment:
.to_string() is not necessary if SASCode derive from PartialOrd
#[derive(Debug, PartialEq, Eq)]
pub struct SASCode(String);